### PR TITLE
Add .multipleLine 'UIHint' to 'RSTTableStepViewController'

### PR DESCRIPTION
**Pull request by joshbruhin**
- In 'RSTTableStepViewController', add support for new 'RSDStepTextViewCell' and quite a bit of refactoring to support various behaviors for both the 'RSDStepTextViewCell' and 'RSDStepTextFieldCell', such as scrolling the cell visible when it becomes first responder.

- Implement new 'RSDStepTextViewCell' class; Add constants protocol and struct for the text view class; Add new 'RSDStepTextInputView' protocol to provide access to common properties for the 'RSDStepTextView' and 'RSDStepTextField' classes

- Add color and font definitions for new custom text view cell.

### CODE REVIEW

**Review by syoung-smallwisdom — APPROVED** [review]
looks good

File: ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift [hunk 1]
```diff
@@ -344,23 +344,162 @@ open class RSDStepTextFieldCell: RSDTableViewCell {
...
+/// form step table view.
+open class RSDStepTextViewCell: RSDTableViewCell {
+    
+    /// The text field associated with this cell.
+    public var textView: RSDStepTextView!
+    
+    /// The label used to display the prompt for the input field.
+    open var viewLabel: UILabel!
+    
+    /// Layout constants. Subclasses can override to customize; otherwise the default private
```
Comment (Erin-Mounts) [inline ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift:357]: They aren't actually private though, are they? Just not writable outside the class itself?

File: ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift [hunk 2]
```diff
@@ -344,23 +344,162 @@ open class RSDStepTextFieldCell: RSDTableViewCell {
...
+public protocol RSDStepTextViewCellLayoutConstants {
+    var verticalMargin: CGFloat { get }
+    var verticalPadding: CGFloat { get }
+    var sideMargin: CGFloat { get }
+    var height: CGFloat { get }
+    var textInset: CGFloat { get }
+    var borderRadius: CGFloat { get }
+}
+
+/// Default constants used by a `RSDStepTextFieldCell`.
```
Comment (Erin-Mounts) [inline ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift:480]: @syoung-smallwisdom  When we generate HTML API docs from the Swift doc comments, do you know if it includes documentation of non-public objects and interfaces like this (or if there's a command line switch to not do that)? It'd be nice to have them documented internally for framework developers, but if these would wind up in the external docs then they should just be plain old comments, not doc comments.

File: ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift [hunk 3]
```diff
@@ -344,23 +344,162 @@ open class RSDStepTextFieldCell: RSDTableViewCell {
...
 fileprivate struct RSDDefaultStepTextFieldCellLayoutConstants {
     let verticalMargin: CGFloat = 10.0
     let verticalPadding: CGFloat = 7.0
     let sideMargin: CGFloat = 24.0
 }
 
+/// Default constants used by a `RSDStepTextViewCell`.
```
Comment (Erin-Mounts) [inline ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift:487]: See above comment about doc comments and private objects

File: ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift [hunk 4]
```diff
@@ -395,9 +534,38 @@ open class RSDStepTextFieldFeaturedCell: RSDStepTextFieldCell {
...
+/// `RSDStepTextInputView` defines custom properties associated with a 'UITextView' or 'UITextField'
+/// and provides read only access to other common properties.
+public protocol RSDStepTextInputView: class {
+    var inputAccessoryView: UIView? { get }
+    var inputView: UIView? { get }
+    var currentText: String? { get set }
+    var indexPath: IndexPath? { get set }
+}
+
+/// `RSDStepTextField` is a subclass of `UITextField` that conforms to 'RSDStepTextInputView'
```
Comment (Erin-Mounts) [inline ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift]: needs a period at the end.

File: ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift [hunk 5]
```diff
@@ -395,9 +534,38 @@ open class RSDStepTextFieldFeaturedCell: RSDStepTextFieldCell {
...
+            return self.text
+        }
+        set {
+            self.text = currentText
+        }
+    }
+    public var indexPath: IndexPath?
+}
+
+/// `RSDStepTextView` is a subclass of `UITextView` that conforms to 'RSDStepTextInputView'
```
Comment (Erin-Mounts) [inline ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift]: period.

**Review by Erin-Mounts — COMMENTED** [review]
Couple of nits/questions, otherwise looks good

File: ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift (hunk 2 above)
Comment (syoung-smallwisdom) [reply ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift:480]: No, right now it is set to only show doc comments for public. private, fileprivate, and internal are all hidden. That said, I will often use a comment on those functions on the assumption that a developer making changes may make them public in a future release.

**Review by syoung-smallwisdom — CHANGES_REQUESTED** [review]
@joshbruhin 
Please see Erin's comments about documentation. Thanks!

We use the `///` comments to mark properties and functions that are or may become public (and thus should be documented). Comments should be structured as sentences with proper punctuation.

We use `//` to mark comments within a method. Ideally, these should also include a period at the end of the sentences. This helps with refactoring when the code changes and part of a method gets refactored into a publicly exposed method (usually to support subclass override).

File: ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift (hunk 1 above)
Comment (joshbruhin) [reply ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift:357]: Right.

File: ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift (hunk 2 above)
Comment (joshbruhin) [reply ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift:480]: @syoung-smallwisdom so should I change "///" to "//" on these 'fileprivate struct' declarations?

Comment (syoung-smallwisdom) [reply ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift:480]: No, it's fine to use `///` and in fact, that's what I will use on the assumption that future development may call for changing the privacy.

**Comment by joshbruhin** [comment]
@syoung-smallwisdom  Nits fixed. Ready to merge

**Review by syoung-smallwisdom — APPROVED** [review]

**Pull request merged**